pyccel/parser/semantic.py

```diff
@@ -1140,57 +1163,120 @@ def _handle_function(self, expr, func, args, is_method = False):
 
             parent_assign = expr.get_direct_user_nodes(lambda x: isinstance(x, Assign) and not isinstance(x, AugAssign))
 
-            func_args = func.arguments if isinstance(func, FunctionDef) else func.functions[0].arguments
             func_results = func.results if isinstance(func, FunctionDef) else func.functions[0].results
-
             if not parent_assign and len(func_results) == 1 and func_results[0].var.rank > 0:
                 tmp_var = PyccelSymbol(self.scope.get_new_name())
                 assign = Assign(tmp_var, expr)
                 assign.set_current_ast(expr.python_ast)
                 self._additional_exprs[-1].append(self._visit(assign))
                 return self._visit(tmp_var)
 
-            if len(args) > len(func_args):
+            func_args = func.arguments if isinstance(func,FunctionDef) else func.functions[0].arguments
+            if not len(args) > len(func_args):
+                args     = self._sort_function_call_args(func_args, args)
+                new_expr = FunctionCall(func, args, self._current_function)
+                for a, f_a in zip(new_expr.args, func_args):
+                    if f_a.persistent_target:
+                        assert is_method
+                        val = a.value
+                        if isinstance(val, Variable):
+                            a.value.is_target = True
+                            self._indicate_pointer_target(args[0].value, a.value, expr)
+                        else:
+                            errors.report(f"{val} cannot be passed to function call as target. Please create a temporary variable.",
+                                    severity='error', symbol=expr)
+
+                if None in new_expr.args:
+                    errors.report("Too few arguments passed in function call",
+                            symbol = expr,
+                            severity='error')
+                elif isinstance(func, FunctionDef):
+                    self._check_argument_compatibility(args, func_args,
+                                expr, func.is_elemental)
+            else:
```

Comment:
there was a deleted code, which is why it looks like this

Reply:
The changes is due to removing a chunk of the code and putting in the function `_sort_function_call_args`

Reply:
The reason it is so hard to see the changes is not because code is deleted but because it is reordered. Before it was:
```python
            if len(args) > len(func_args):
                 ERROR
            .....
```
But now it is
```python
            if len(args) <= len(func_args):
                .....
            else:
                 ERROR
```

Reply:
ah yes , you are right, but I still think this looks better
```python 
            if len(args) <= len(func_args):
                .....
            else:
                 ERROR
```

Reply:
I don't have a strong opinion on which looks better (though I'm slightly inclined to agree with you). However Yaman sometimes asks for line lengths to be kept under 80 characters. This means that excessive indenting leads to lots more line wrapping. Further all the recent interns seem to prefer the version which only uses an if block for the return/raise statement and then continues so this now matches a lot of the code elsewhere in the code base.
Given that this PR is quite complicated and hard to follow I would rather keep unnecessary/aesthetic changes to a minimum.

Reply:
[bd593e5](https://github.com/pyccel/pyccel/pull/1728/commits/bd593e50538c77e3943b955921c0690d0866d708)

Reply:
The `else` still keeps the indenting and doesn't fix the conflicts. It just means they're easier to spot because they're next to each other now

Reply:
ah yes, you are right, there is no need for `else`

Reply:
Thanks